core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RangeValueFilterExecuterImpl.java

```diff
@@ -0,0 +1,557 @@
+    BitSet bitSet = new BitSet(numerOfRows);
+    int start = 0;
+    int startIndex = 0;
+    int startMin = 0;
+    int endMax = 0;
+    byte[][] filterValues = this.filterRangesValues;
+
+    // For Range expression we expect two values. The First is the Min Value and Second is the
+    // Max value.
+    // Get the Min Value
+    start = CarbonUtil
+        .getFirstIndexUsingBinarySearch(dimensionColumnDataChunk, startIndex, numerOfRows - 1,
+            filterValues[0], false);
+    if (isGreaterThan() == true && start >= 0) {
+      start = CarbonUtil
+          .nextGreaterValueToTarget(start, dimensionColumnDataChunk, filterValues[0], numerOfRows);
+    }
+
+    if (start < 0) {
+      start = -(start + 1);
+      if (start == numerOfRows) {
+        start = start - 1;
+      }
+      // Method will compare the tentative index value after binary search, this tentative
+      // index needs to be compared by the filter member if its >= filter then from that
+      // index the bitset will be considered for filtering process.
+      if ((isGreaterThan() == true) && (ByteUtil.compare(filterValues[0],
+          dimensionColumnDataChunk.getChunkData(dimensionColumnDataChunk.getInvertedIndex(start))))
+          > 0) {
+        start = start + 1;
+      } else if ((isGreaterThanEqualTo() == true) && (ByteUtil.compare(filterValues[0],
+          dimensionColumnDataChunk.getChunkData(dimensionColumnDataChunk.getInvertedIndex(start))))
+          >= 0) {
+        start = start + 1;
+      }
+    }
+    startMin = start;
+
+    // Get the Max value
+    start = CarbonUtil
+        .getFirstIndexUsingBinarySearch(dimensionColumnDataChunk, startIndex, numerOfRows - 1,
+            filterValues[1], false);
+    if (isLessThan() == true && start >= 0) {
+      start = CarbonUtil.nextLesserValueToTarget(start, dimensionColumnDataChunk, filterValues[1]);
+    }
+
+    if (start < 0) {
+      start = -(start + 1);
+      if (start == numerOfRows) {
+        start = start - 1;
+      }
+      // Method will compare the tentative index value after binary search, this tentative
+      // index needs to be compared by the filter member if its >= filter then from that
+      // index the bitset will be considered for filtering process.
+      if ((isLessThan() == true) && (ByteUtil.compare(filterValues[1],
+          dimensionColumnDataChunk.getChunkData(dimensionColumnDataChunk.getInvertedIndex(start)))
+          < 0)) {
+        start = start - 1;
+      } else if ((isLessThanEqualTo() == true) && (ByteUtil.compare(filterValues[1],
+          dimensionColumnDataChunk.getChunkData(dimensionColumnDataChunk.getInvertedIndex(start)))
+          <= 0)) {
+        start = start - 1;
+      }
+    }
+    endMax = start;
+
+    for (int j = startMin; j <= endMax; j++) {
+      bitSet.set(dimensionColumnDataChunk.getInvertedIndex(j));
+    }
+
+    if (dimensionColumnDataChunk.isNoDicitionaryColumn()) {
```

Comment:
Binary search cannot be done on "@NU#LL$!", so need to check and compare for null on matching rows.